src/sentry/api/utils.py

```diff
@@ -266,3 +267,20 @@ def customer_domain_path(path: str) -> str:
         if updated != path:
             return updated
     return path
+
+
+def method_dispatch(**table):
+    """
+    Dispatches a incoming request to a different handler based on the HTTP method
+
+    >>> url('^foo$', method_dispatch(POST = post_handler, GET = get_handler)))
```

Comment:
Don't endpoint classes already do this? Or are you looking to migrate view functions?

Reply:
I am trying to split apart two monitor endpoints that live on the same URL. Monitor ingestion happens via `POST /monitor/<id>/checkins` and listing checkins happens via `GET /monitor/<id>/checkins` but those two endpoints have totally different types of authentication and have different requirements for their `convert_args`

Reply:
(already discussed on slack, just here for posterity)